src/Raven.Client/Util/AsyncHelpers.cs

```diff
@@ -37,7 +37,15 @@ public static void RunSync(Func<Task> task)
             if (UseTaskAwaiterWhenNoSynchronizationContextIsAvailable && oldContext == null)
             {
                 // We can run synchronously without any issue.
-                task().GetAwaiter().GetResult();
+                try
+                {
+                    task().GetAwaiter().GetResult();
+                }
+                catch (AggregateException ex)
+                {
+                    var exception = ex.ExtractSingleInnerException();
+                    ExceptionDispatchInfo.Capture(exception).Throw();
```

Comment:
What if there is not a single exception here?

Reply:
ExtractSingleInnerException handles that.
It will return the origin (aggregate) exception.

```
public static Exception ExtractSingleInnerException(this AggregateException e)
{
    if (e == null)
        return null;
    while (true)
    {
        if (e.InnerExceptions.Count != 1)
            return e;

        var aggregateException = e.InnerExceptions[0] as AggregateException;
        if (aggregateException == null)
            break;
        e = aggregateException;
    }

    return e.InnerExceptions[0];
}
```

Reply:
Returns same ex, should we also tackle here TimeoutException and use stopwatch?

From code below:
```
                var exception = ex.ExtractSingleInnerException();
                if (exception is OperationCanceledException)
                    throw new TimeoutException("Operation timed out after: " + sw.Elapsed, ex);
                ExceptionDispatchInfo.Capture(exception).Throw();
```

Reply:
Now that you say that maybe better to keep the origin exception.
It smells like there was aggregation because of the syn execution
I couldn't get such results when I tested it on dotnet 3.1 or any of the dotnet framework.
Maybe better to keep the origin exception?
In the test the aggregate exception is explicit - 
https://github.com/ravendb/ravendb/blob/551f6a9e513b76d5a0493df90111ab142e88e6b3/src/Sparrow/Threading/DisposeOnce.cs#L255

Reply:
Having a different exception type depending on the code path is sth that I would like to avoid

Reply:
Ping me tomorrow

Reply:
@haludi Please make sure we have the same handling in all places where we `catch (AggregateException ex)`. Can we introduce a single method handling that?

Reply:
Addressed by @ppekrol in https://github.com/ravendb/ravendb/pull/18188/commits/dfb3e5ca886c9c15b35558540b8dc34cb4c994f1